rct229/rulesets/ashrae9012019/section11/section11rule15.py

```diff
@@ -0,0 +1,213 @@
+                    )
+                )
+                swh_use_ids.extend(service_water_heating_use_ids_b)
+
+            for building_segment in find_all(
+                "$.buildings[*].building_segments[*]", rmd_p
+            ):
+                service_water_heating_use_ids_p = (
+                    get_swh_uses_associated_with_each_building_segment(
+                        rmd_p, building_segment["id"]
+                    )
+                )
+                swh_use_ids.extend(service_water_heating_use_ids_p)
+            return len(set(swh_use_ids)) > 0
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+            swh_dist_sys_dict_b = {}
+            swh_dist_sys_dict_p = {}
+            for swh_dist_sys_b in find_all(
+                "$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*]",
+                rmd_b,
+            ):
+                swh_dist_sys_dict_b[swh_dist_sys_b["id"]] = swh_dist_sys_b
+            for swh_dist_sys_p in find_all(
+                "$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*]",
+                rmd_p,
+            ):
+                swh_dist_sys_dict_p[swh_dist_sys_p["id"]] = swh_dist_sys_p
+            return {
+                "swh_dist_sys_dict_b": swh_dist_sys_dict_b,
+                "swh_dist_sys_dict_p": swh_dist_sys_dict_p,
+            }
+
+        class SWHUseRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section11Rule15.RMDRule.SWHUseRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                )
+
+            def get_calc_vals(self, context, data=None):
+                swh_dist_sys_dict_b = data["swh_dist_sys_dict_b"]
+                swh_dist_sys_dict_p = data["swh_dist_sys_dict_p"]
+                rule_status = "pass"
+                rule_note = ""
+                manual_check_msg = ""
+                swh_use_b = context.BASELINE_0
+                swh_use_p = context.PROPOSED
+                if not swh_use_b and swh_use_p:
+                    rule_status = "fail"
+                    rule_note = (
+                        rule_note
+                        + swh_use_b["id"]
+                        + " exists in the proposed model, but not in the baseline."
+                    )
+                elif swh_use_b and not swh_use_p:
+                    rule_status = "fail"
+                    rule_note = (
+                        rule_note
+                        + swh_use_b["id"]
+                        + " exists in the baseline, but not in the proposed model."
+                    )
+                else:
+                    if swh_use_b.get("use_units") != swh_use_p.get("use_units"):
+                        rule_status = "fail"
```

Comment:
there are two more fields in the latest schema [is_heat_recovered_by_drain](https://github.com/pnnl/ruleset-checking-tool/blob/73427d60cbc3b6a12ee35756ca9d6240defc9e1c/rct229/schema/ASHRAE229.schema.json#L4785) and [is_recovered_heat_used_by_cold_side_feed](https://github.com/pnnl/ruleset-checking-tool/blob/73427d60cbc3b6a12ee35756ca9d6240defc9e1c/rct229/schema/ASHRAE229.schema.json#L4790)

Reply:
Yes, they are boolean. My understanding is that their equality determines if the swh uses in the baseline and proposed are modeled the same. For example, if [is_heat_recovered_by_drain](https://github.com/pnnl/ruleset-checking-tool/blob/73427d60cbc3b6a12ee35756ca9d6240defc9e1c/rct229/schema/ASHRAE229.schema.json#L4785) is true in the baseline and false in the proposed, the energy required by this swh use will be different in the two models.

Reply:
Addressed